### PR TITLE
feature Integrate Shoryuken into ddtracer

### PR DESCRIPTION
**Pull request by JustSnow**

### CODE REVIEW

**Comment by JustSnow** [comment]
@delner fixed comments. Will think about tests. If you have suggestions, would be good to see them :)

**Comment by delner** [comment]
@JustSnow For tests, I think we just need some basic RSpec tests that verify that it traces a job properly. You'd probably want to find a way to mock an incoming message from SQS (from within the test suite as opposed to setting up an external service), then use that to drive the test and verify it generated traces with the expected attributes. You might want to look at something like Racecar as an example.

Also, could you try this out in an actual application and share a screenshot of how the trace looks in the UI? I'd very much like to see what it looks like in action!

**Comment by JustSnow** [comment]
@delner 
shoryuken-datadog-trace

**Comment by delner** [comment]
@JustSnow Overall this is looking very good; I think its all configured pretty much as we'd like to see it. We should add more tests that simulate a Shoryuken job running, which verify that it produces a span with the appropriate name, resource, service, and tags w/ values at minimum.

There should be a number of other integrations you should be able to copy tests from that verify similar behavior, it'd just be a matter then of modifying them to have them trigger a Shoryuken job.

Once we have that in, I think we can do a final review and merge this.

**Comment by JustSnow** [comment]
@delner totally agree. working on it, but don't have enough time to finish tests

File: spec/ddtrace/contrib/shoryuken/tracer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      queue_url: queue,
+      body: 'test',
+      attributes: {
+        id: 1,
+        test_attribute: 'test'
+      }
+  end
+
+  before do
+    allow(Shoryuken::Client).to receive(:queues).with(queue).and_return sqs_queue
```
Comment (JustSnow) [inline spec/ddtrace/contrib/shoryuken/tracer_spec.rb]: @delner maybe do you know or some one can suggest right way how to run shoryuken client for test to getting spans from writer

**Comment by delner** [comment]
@JustSnow I created a [diff.txt](https://github.com/DataDog/dd-trace-rb/files/2654179/diff.txt) you can apply to this PR which should update it with a test that runs Shoryuken in a test. (Your repo didn't grant push permissions from the upstream, otherwise I would've pushed it myself.)

Unfortunately it appears our tests for this won't work though for this because [Shoryuken doesn't run middleware when running workers inline for tests](https://github.com/phstc/shoryuken/issues/541). I opened an issue and we'll see if this gets merged/released for Shoryuken.

We probably can merge and release this integration in the mean time, but we'll need to make sure we update the CI accordingly later.

**Comment by JustSnow** [comment]
@delner updated according `diff.txt`. Hope all are right. This is strange what you can't push commits here, because I allowed to do it all who has edit permission on dd-trace repo (`Allow edits from maintainers.` is enabled)

**Pull request merged**